### PR TITLE
[docs] Update on `renderCell` & autogenerated rows

### LINKED ISSUE #8933: [docs] Update `renderCell`
### Duplicates

- [X] I have searched the existing issues

### Related page

https://mui.com/x/react-data-grid/column-definition/#rendering-cells

### Kind of issue

Unclear explanations

### Issue description

- Have a more meaningful example (maybe a Rating component?)
- Explain that `renderCell` is also being used for autogenerated rows and show a way to opt out of using `renderCell` for autogenerated rows

### Context 🔦

_No response_

### PR DESCRIPTION
**Pull request by romgrk**
Closes #8933 

Clarifies how to handle autogenerated rows in the column definition functions.

Preview: https://deploy-preview-13879--material-ui-x.netlify.app/x/react-data-grid/column-definition/#autogenerated-rows

### CODE REVIEW

**Review by flaviendelangle — COMMENTED** [review]
:pray: This is a doc section that could have been done years ago

File: docs/data/data-grid/column-definition/column-definition.md [hunk 1]
```diff
@@ -317,6 +332,26 @@ In the following example, the "Delete" action opens a confirmation dialog and th
...
+
+Some features like [row grouping](/x/react-data-grid/row-grouping/) or [aggregation](/x/react-data-grid/aggregation/) create autogenerated rows.
+These rows also call functions like `valueGetter`, `valueFormatter` and `renderCell`, and that can cause issues if you're not expecting it because the `row` parameter will be an empty object and the `value` parameter will be `undefined`.
+If we take for example the movie dataset, you can detect autogenerated rows like this:
+
+```tsx
+{
+  field: 'title',
+  valueGetter: (value, row) => {
+    if (String(apiRef.current.getRowId(row)).startsWith('auto-generated-row-')) {
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply docs/data/data-grid/column-definition/column-definition.md]: > Also the codebase can't seem to agree on "autogenerated" or "auto-generated", I've seen both in a few places.

That one is sad :laughing:

File: docs/data/data-grid/column-definition/AutogeneratedRows.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const initialState = useKeepGroupedColumnsHidden({
+    apiRef,
+    initialState: {
+      rowGrouping: {
+        model: ['company'],
+      },
+    },
+  });
+
+  const columns: GridColDef[] = [
```
Comment (cherniavskii) [inline docs/data/data-grid/column-definition/AutogeneratedRows.tsx:24]: Nitpick: `columns` can be defined outside of the component

File: docs/data/data-grid/column-definition/AutogeneratedRows.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  DataGridPremium,
+  GridColDef,
+  isAutogeneratedRow,
+  useGridApiRef,
+  useKeepGroupedColumnsHidden,
+} from '@mui/x-data-grid-premium';
+import { useMovieData } from '@mui/x-data-grid-generator';
+
+export default function AutogeneratedRows() {
+  const data = useMovieData();
```
Comment (cherniavskii) [inline docs/data/data-grid/column-definition/AutogeneratedRows.tsx:12]:
```suggestion
  const { rows } = useMovieData();
```

To make it more explicit that it's only used for generating rows.

**Review by cherniavskii — APPROVED** [review]
Thanks for taking this one!

**Pull request merged**